connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -928,10 +929,10 @@ object SparkSession extends Logging {
      *
      * @since 3.5.0
      */
-    def config(map: Map[String, Any]): Builder = synchronized {
+    def config(map: Map[String, Any]): Builder = {
       map.foreach { kv: (String, Any) =>
         {
-          options += kv._1 -> kv._2.toString
+          options.put(kv._1, kv._2.toString)
```

Comment:
This change seems to be a breaking change as another thread can read/write between each `put`, whereas can't before? How about using `putAll` instead?